src/main/java/lib/Injector.java

```diff
@@ -0,0 +1,29 @@
+package lib;
+
+import dao.BetDao;
+import dao.UserDao;
+import factory.DaoFactory;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+public class Injector {
+    public static Object getInstance(Class clazz) throws NoSuchMethodException,
+            IllegalAccessException, InvocationTargetException, InstantiationException {
+        Constructor constructor = clazz.getDeclaredConstructor();
+        Object instance = constructor.newInstance();
+        Field[] declaredFields = clazz.getDeclaredFields();
+        for (Field field : declaredFields) {
+            if (field.getAnnotation(Inject.class) != null) {
+                field.setAccessible(true);
+                if (field.getType().equals(BetDao.class)) {
+                    field.set(instance, DaoFactory.getBetDao());
+                }
+                if (field.getType().equals(UserDao.class)) {
+                    field.set(instance, DaoFactory.getUserDao());
+                }
```

Comment:
let's refactor lines 19-24

1. check if the field's type is BetDao and there is `@Dao` in BetDaoImpl - then set field
2. else - check if the field's type is UserDao and there is `@Dao` in UserDaoImpl - then set field
3. else - throw your custom Exception

Reply:
done